pkg/vsphere/disk/disk_manager.go

```diff
@@ -281,7 +288,9 @@ func (m *Manager) Attach(op trace.Operation, disk *types.VirtualDisk) error {
 	machineSpec := types.VirtualMachineConfigSpec{}
 	machineSpec.DeviceChange = append(machineSpec.DeviceChange, changeSpec...)
 
-	m.reconfig.Lock()
+	m.mu.Lock()
+	defer m.mu.Unlock()
```

Comment:
Correct, the locking and consistency of the pkg is not something I really liked. There are two locks there, manager has a reconfigure lock which serializes reconfigure operations, and disks have a mutex to serialize attach/detach operations.

The way we use those locks are confusing. Disk.go uses locks for some of the exported functions but manager mostly calls non-exported without holding any lock. Will try to address this, stay tuned :)